### ISSUE TITLE
staging files is slow for large repos

### ISSUE THREAD

**Issue by dasl-** [jesseduffield/lazygit#1605]
**Describe the bug**
When I stage a file by pressing A or SPACE, it takes 1 to 3 seconds for the file to become staged. The repo I am working on is large: 430,000 commits and 15,000 branches.

**Desktop (please complete the following information):**
 - OS: Centos `Linux <redacted> 3.10.0-862.3.2.el7.x86_64 #1 SMP Mon May 21 23:36:36 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`
 - Lazygit Version 0.31.4
 - The last commit id 02bf6a5c177c145c46e5e62524458fc3375d01af

**Comment by mark2185** [comment]
Does the same happen when you add it manually through `git`?

**Comment by mark2185** [comment]
Could you test current `master`? I do believe #2072 should help with this exact use case.

**Comment by ponsheng** [comment]
@mark2185, I had the same issue with version 0.35 on Linux.
And it got resolved with current master (36c6462a534ff328051bd81ed90eb626d1a59f09).
Thanks.

**Comment by liancheng** [comment]
TL;DR: Set `status.showUntrackedFiles` to `normal` or `no` resolved this issue for me.

I hit this issue recently. By running `lazygit --debug` and `lazygit --logs`, I saw that LazyGit took 10 seconds to run `git status`. The concrete command line was:

```
Sep 18 13:17:45 |INFO| git status --untracked-files=all --porcelain -z (10.577895417s)
```

Running the same command with `--untracked-files=normal` was significantly faster. By reading LazyGit source code, I found that this is controlled by the standard Git `status.showUntrackedFiles` config. Setting it to `normal` globally or locally in my large repo resolved my performance issue.

**Comment by ChrisMcD1** [comment]
/close

Seems like we have a fix, and a workaround if the fix doesn't work.
